test/wasmfs/wasmfs_fetch.c

```diff
@@ -150,6 +151,108 @@ void test_small_reads() {
   assert(close(fd) == 0);
 }
 
+void test_small_chunks() {
+  // Read the file in small amounts.
+  printf("Running %s...\n", __FUNCTION__);
+
+  char expected[] = "hello";
+  size_t size = 5;
+
+  backend_t backend = wasmfs_create_fetch_backend("small.dat",2);
+  int fd;
+  char buf[size + 1];
+  fd = wasmfs_create_file("/testfile4", 0777, backend);
+  for (size_t i = 0; i < size; i+=1) {
+    int read_now = read(fd, buf + i, 1);
+    assert(read_now <= 1);
+    printf("read some bytes smaller than chunk size\n");
+  }
+  buf[size] = 0;
+  printf("buf %s\n",buf);
+  assert(strcmp(buf, "hello") == 0);
+
+  assert(close(fd) == 0);
+
+  fd = wasmfs_create_file("/testfile5", 0777, backend);
+  for (size_t i = 0; i < size; i+=2) {
+    int read_now = read(fd, buf + i, 2);
+    assert(read_now <= 2);
+    printf("read some bytes equal to chunk size\n");
+  }
+  buf[size] = 0;
+  printf("buf %s\n",buf);
+  assert(strcmp(buf, "hello") == 0);
+
+  assert(close(fd) == 0);
+
+  fd = wasmfs_create_file("/testfile6", 0777, backend);
+  for (size_t i = 0; i < size; i+=5) {
+    int read_now = read(fd, buf + i, 5);
+    assert(read_now <= 5);
+    printf("read some bytes much larger than chunk size\n");
+  }
+  buf[size] = 0;
+  printf("buf %s\n",buf);
+  assert(strcmp(buf, "hello") == 0);
```

Comment:
How about making a helper function for these read loops, that receives the chunk size?

Reply:
Good idea